internal/rbd/controllerserver.go

```diff
@@ -156,7 +186,13 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	}
 	defer cs.VolumeLocks.Release(req.GetName())
 
-	found, err := checkVolExists(ctx, rbdVol, cr)
+	err = rbdVol.Connect(cr)
```

Comment:
At this time,`rbdvol.Connect` and cluster connection  do same thing. But I believe the idea is to have the rbbvol.Connect to have more functionlities in the future. So one question is, what exactly going to be  part of rbdvol.Connect()? @nixpanic

Reply:
The cluster connection is a private component of the rbdVolume. In order to be able to clean up the contents of the packages, we need to make a clearer distinction what API's are private and which are public. Ideally the nodeserver and controllerserver do not need to access any internals from the rbd package, but only the rbd package's public API. The Connect() method is part of cleaning this up.

Reply:
Sure. but my query was mainly on, does rbdvol is a "connect" to a 'specific' volume or its going to a handle to all RBD volumes in the Pool ? @nixpanic

Reply:
`Connect()` will build the (shared) connection to the Ceph cluster. Depending on the functions that are executed, there is an `IOContext` needed, which is specific to a pool. Everything will go over the shared connection in the end (not I/O, only the image create/delete/... operations).

Reply:
yeah, that matches my expectation. If that's the case, do you think a better name would be `rbdpool.Connect` or `rbdhandle.Connect` which can be embedded in rbdVol struct field? just a thought though. @nixpanic

Reply:
It was suggested to not embed the connection in the rbdVolume, so I removed the embedding. The rbdVolume still needs to connect to the cluster, so I called the method `Connect()`. If that is confusing, I could rename the method to `ConnectToCluster()` or such.

Reply:
IMO, one thing which we have to narrow down here is :

 Do we need rbdvolume to connect to cluster or not ? if yes, embedding a connector in the rbdvol looks a decent solution to me. If we do that, what we actually need? cluster connection or a connection-specific to list or operate only `rbd` commands or a connection which can operate only this specific Volume?

If its about cluster connection, we can embed cluster connector itself. If its about specific volume, I dont think we need check in later stage of this code path like `rbdVol.Exist()` ..etc. if its about the permission to operate on `any` RBD volume in the cluster, may be this connection has to be established by a specific credential that has limited permissions..etc, not sure its a valid use case though. @nixpanic

Reply:
Currently an rbdVolume has a single connection. In the future we may want to split the connection, and have shared+cached IOContext objects instead (or in addition). Depending on what connections an rbdVolume needs (different IOContexts for different pools, separate journalling, topology-aware, multi-site, ...), embedding a connection object is not sufficient.

Reply:
> Currently an rbdVolume has a single connection. In the future we may want to split the connection, and have shared+cached IOContext objects instead (or in addition). Depending on what connections an rbdVolume needs (different IOContexts for different pools, separate journalling, topology-aware, multi-site, ...), embedding a connection object is not sufficient.

but if this is about supporting above mentioned features, this connection object is basically a pool connection. Need some thoughts on this. Lets see.

Meanwhile, lets get this in and do the improvements accordingly. 

Approved.  Thanks @nixpanic